homeassistant/components/bluetooth/manager.py

```diff
@@ -483,15 +483,21 @@ def _async_remove_callback() -> None:
         # immediately with the last packet so the subscriber can see the
         # device.
         all_history = self._get_history_by_type(connectable)
-        if (
-            (address := callback_matcher.get(ADDRESS))
-            and (service_info := all_history.get(address))
-            and ble_device_matches(callback_matcher, service_info)
-        ):
-            try:
-                callback(service_info, BluetoothChange.ADVERTISEMENT)
-            except Exception:  # pylint: disable=broad-except
-                _LOGGER.exception("Error in bluetooth callback")
+        service_infos: Iterable[BluetoothServiceInfoBleak]
+        if address := callback_matcher.get(ADDRESS):
+            if service_info := all_history.get(address):
+                service_infos = [service_info]
+            else:
+                service_infos = []
+        else:
+            service_infos = all_history.values()
```

Comment:
```suggestion
        service_infos: Iterable[BluetoothServiceInfoBleak] = []
        if address := callback_matcher.get(ADDRESS):
            if service_info := all_history.get(address):
                service_infos = [service_info]
        else:
            service_infos = all_history.values()
```

Any reason to not initialize it to an empty list?

Reply:
just that it will be overwritten in just about all cases. But sure it's one line shorter :)